Dockerfile

```
@@ -0,0 +1,60 @@
+ARG PYTHON_VERSION_CONSTRAINT
+FROM python:${PYTHON_VERSION_CONSTRAINT}-slim-bookworm as python-base
+
+LABEL maintainer="moneymeets GmbH <admin@moneymeets.com>"
+
+ENV PYTHONUNBUFFERED=1 \
+    PYTHONDONTWRITEBYTECODE=1 \
+    APP_PATH="/opt/app" \
+    POETRY_HOME="/opt/poetry" \
+    POETRY_NO_INTERACTION=1 \
+    POETRY_VIRTUALENVS_IN_PROJECT=true \
+    VENV_PATH="/opt/app/.venv"
+
+ENV PATH="$POETRY_HOME/bin:$VENV_PATH/bin:$PATH"
+
+ARG POETRY_VERSION
+FROM python-base as builder-deps
+
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt install -y curl git  \
+    && apt-get install -y --no-install-recommends build-essential openssh-client \
+    && curl -sSL https://install.python-poetry.org | python - --yes --version=${POETRY_VERSION} \
+    && apt remove -y --purge curl \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY .git/ ./.git/
+
+RUN git archive -v -o app.tar.gz --format=tar.gz HEAD
+
+WORKDIR $APP_PATH
+
+ADD poetry.lock pyproject.toml ./
+
+RUN --mount=type=ssh --mount=type=cache,target=/root/.cache \
+    mkdir -p -m 0600 ~/.ssh && \
+    ssh-keyscan github.com >> ~/.ssh/known_hosts && \
+    poetry install --without dev
```

We don't need the ssh stuff, because we are not using private dependencies.
```suggestion
RUN --mount=type=cache,target=/root/.cache \
    poetry install --without dev
```